python/pyarrow/io.pxd

```diff
@@ -25,6 +25,8 @@ from pyarrow.includes.libarrow_io cimport (ReadableFileInterface,
 cdef class Buffer:
     cdef:
         shared_ptr[CBuffer] buffer
+        Py_ssize_t shape[1]
+        Py_ssize_t strides[1]
```

Comment:
Does this need to be in the struct? It looks like you copy them into the buffer below